pages/onboarding.vue

```diff
@@ -309,54 +319,225 @@ const previousStep = () => {
   }
 };
 
-const skipStep = () => {
-  nextStep();
+const ensureOnboardingComplete = async () => {
+  const completeRes = await configurationsApi
+    .create({
+      key: CONFIGURATION_KEYS.ONBOARDING_COMPLETE,
+      value: { completedAt: new Date().toISOString() },
+      type: 'json'
+    })
+    .catch(() => null);
+  if (!completeRes) {
+    await configurationsApi
+      .update(CONFIGURATION_KEYS.ONBOARDING_COMPLETE, {
+        value: { completedAt: new Date().toISOString() },
+        type: 'json'
+      })
+      .catch(() => null);
+  }
+  if (typeof globalThis !== 'undefined') {
+    globalThis.localStorage.setItem('onboarding-completed', 'true');
+  }
+
+  // Refresh configurations cache to ensure middleware sees the update
+  await sharedData.loadConfigurations(true).catch(() => null);
+};
+
+const skipStep = async () => {
+  console.log('Skip step clicked');
+  try {
+    // Mark onboarding as complete regardless of current step
+    console.log('Calling ensureOnboardingComplete');
+    await ensureOnboardingComplete();
+    console.log('ensureOnboardingComplete finished');
+    // Redirect to dashboard after completion
+    console.log('Redirecting to dashboard');
+    router.push('/dashboard');
+  } catch (e) {
+    console.error('Error in skipStep:', e);
+  }
 };
 
 const handleLanguageSelection = () => {
   if (!selectedLanguage.value) return;
 
-  showSuccess(
-    'Language Set!',
-    `Language changed to ${availableLanguages.find((l) => l.code === selectedLanguage.value)?.name}.`
-  );
+  // Just store locally - will be saved when onboarding completes
+  const langName = availableLanguages.find((l) => l.code === selectedLanguage.value)?.name;
+  showSuccess('Language Set!', `Language changed to ${langName}.`);
   nextStep();
 };
 
 const handleWalletCurrencySetup = () => {
   if (!isWalletCurrencySetupValid.value) return;
 
-  let message = 'Settings saved successfully!';
-
+  // Just store locally - will be saved when onboarding completes
   if (walletChoice.value === 'rename-default') {
-    message = `Default wallet renamed to "${newWalletName.value}" and currency set to ${selectedCurrency.value}.`;
+    showSuccess(
+      'Setup Complete!',
+      `Default wallet renamed to "${newWalletName.value}" and currency set to ${selectedCurrency.value}.`
+    );
   } else if (walletChoice.value === 'create-new') {
-    message = `New wallet "${newWalletName.value}" created with ${newWalletCurrency.value} currency.`;
+    showSuccess(
+      'Setup Complete!',
+      `New wallet "${newWalletName.value}" created with ${newWalletCurrency.value} currency.`
+    );
   } else {
-    message = `Default wallet configured with ${selectedCurrency.value} currency.`;
+    showSuccess(
+      'Setup Complete!',
+      `Default wallet configured with ${selectedCurrency.value} currency.`
+    );
   }
```

Comment:
- So when you store locally for each one and at the end the save fails (attempting to save all configurations) what happens when you already told the user their configurations were saved?

- All a user might skip after setting 1 or two configs (those will be lost)...

Don't lie in the code, save each config as the user sets them and move to the next that is more straight forwared.